File: aula20_08/app/controller/ProcessoController.java
package aula20_08.app.controller;

import java.util.ArrayList;
import java.util.List;

import aula20_08.app.model.ProcessoService;
import aula20_08.app.model.Processo;
import aula20_08.app.view.ProcessoView;

public class ProcessoController {

    private final ProcessoService service;
    private final ProcessoView view;
    private final List<Processo> listaDeProcessos;

    public ProcessoController(ProcessoService service, ProcessoView view) {
        this.service = service;
        this.view = view;
        this.listaDeProcessos = new ArrayList<>();
    }

    public void iniciarSistema() {
        // 1. Executa a regra de negócio para gerar dados
        service.popular(listaDeProcessos, 4);
        
        // 2. Comanda a View para exibir o estado inicial
        view.exibirLista("Exibindo a lista original", listaDeProcessos);

        // 3. Executa a regra de negócio de ordenação
        service.ordenarBolha(listaDeProcessos);

        // 4. Comanda a View para exibir o resultado final
        view.exibirLista("Exibindo a lista ordenada pelo id", listaDeProcessos);
    }

}

File: python/app/controller/processo_controller.py
from model.processo_service import processo_service
from model.processo import processo
from view.processo_view import processo_view

class processo_controller:
    def __init__(self, service: processo_service, view: processo_view):
        self.service = service
        self.view = view
        self.lista_processo = []

    def iniciar_sistema(self):
        self.service.popular(self.lista_processo, 4)
        self.view.exibir_lista("Mostrando a lista com IDs gerados", self.lista_processo)
        self.service.ordenar_bolha(self.lista_processo)
        self.view.exibir_lista("Lista com IDs em ordem", self.lista_processo)